askar-crypto/src/alg/any.rs

```diff
@@ -84,17 +92,28 @@ impl AnyKey {
 /// Create `AnyKey` instances from various sources
 pub trait AnyKeyCreate: Sized {
     /// Generate a new key from a key material generator for the given key algorithm.
-    fn generate(alg: KeyAlg, rng: impl KeyMaterial) -> Result<Self, Error>;
+    fn generate_with_rng(alg: KeyAlg, rng: impl KeyMaterial) -> Result<Self, Error>;
+
+    /// Generate a new key with an id for the given key algorithm.
+    fn generate_with_id(alg: KeyAlg, id: &str) -> Result<Self, Error>;
```

Comment:
This seems to be restricted to hardware backed keys, and the naming doesn't really reflect that for me. Do we need to accept the ID as an argument or could it just be generated automatically?

Reply:
Right now the id is generated in the FFI layer. I could also defer this to `askar-crypto` where I replace `generate_any_with_id` with `generate_any_hardware` and generate the id there. This would also remove the `uuid` dependency from `aries-askar`, but would introduce it into `askar-crypto`.

I will implement it like this, but also open to other suggestions.